src/lib/rest/RestService.cpp

```diff
@@ -52,9 +52,111 @@
 #define MAX_TENANT_NAME_LEN_STRING    "50"
 
 
+static std::string textParseAttributeValue(ConnectionInfo* ciP, ContextAttribute* caP)
+{
+  double d;
+
+  // 1. Starts and ends with citation marks?
+  if (ciP->payload[0] == '"')
+  {
+    char* end = &ciP->payload[strlen(ciP->payload) - 1];
+
+    if (*end == '"')
+    {
+      *end = 0;
+      caP->stringValue = &ciP->payload[1];
+      caP->valueType   = orion::ValueTypeString;
+    }
+    else
+    {
+      OrionError oe(SccBadRequest, "Missing citation-mark at end of string");
+
+      return oe.render(ciP, "");
+    }
+  }
+
+  // 2. True or false?
+  else if ((strlen(ciP->payload) == 4) && ((strcmp(ciP->payload, "true") == 0) || (strcmp(ciP->payload, "True") == 0) || (strcmp(ciP->payload, "TRUE") == 0)))
+  {
+    caP->boolValue   = true;
+    caP->valueType   = orion::ValueTypeBoolean;
+  }
+  else if ((strlen(ciP->payload) == 5) && ((strcmp(ciP->payload, "false") == 0) || (strcmp(ciP->payload, "False") == 0) || (strcmp(ciP->payload, "FALSE") == 0)))
+  {
+    caP->boolValue   = false;
+    caP->valueType   = orion::ValueTypeBoolean;
+  }
+
+  // 3. Null ?
+  else if ((strlen(ciP->payload) == 4) && ((strcmp(ciP->payload, "null") == 0) || (strcmp(ciP->payload, "Null") == 0) || (strcmp(ciP->payload, "NULL") == 0)))
+  {
+    caP->valueType   = orion::ValueTypeNull;
+  }
+
+  //
+  // 4. Valid Double?
+  // FIXME P4: this is much more complex than just (atof(string) != 0 || string == "0")
+  //           0.000 also is a valid float and it given 0 - 0.0000000000, and 0e0 also ...
+  //           And, even worse, 123K gives 123.0 back, we would need to analyze the string to try to
+  //           find garbage bytes after it if we want to detect this error.
+  //           However, all of this is not so extremely important and for now, (atof(string) != 0 || string== "0") is good enough
+  //
+  else if (((d = strtod(ciP->payload, NULL)) != 0) || ((ciP->payload[0] == '0') || (ciP->payload[1] == 0)))
```

Comment:
Yeah, atof and strtod is pretty much the same. The comment was just for a broader understanding, valid for atof, atod, atoi, strtod, strtoul, etc.  I will change it, to strtod, of course. Unnecessary confusion is always bad.

About

    ((ciP->payload[0] == '0') || (ciP->payload[1] == 0)) ...

... this is the fastest way to do

    strcmp(ciP->payload, "0") == 0

The first char in the char* is '0' (ASCII 48) and the second char is 0  (string terminator), thus "0".

The problem with this part is the difficulty to decide that it is a valid float when strtod(XXX) returns 0.

Actually, now that I think of it ... even if it returns != 0, the problem is still there ...
 
strtod("3x") returns 3.0, as far as I know ...

*I do know that atoi works this way, even did a minor demonstration not long ago. Makes sense to assume strtod works the very same way*

So, examples of 'OK when 0':

    0
    0.00000
    0e2  // Not sure about this syntaxis - I have nerve used it !!!

Examples of 'NOT OK when 0', and an error should be returned:

    Hi
    0x

So, my very simple solution is to assume that, if strtod gives back zero and the string is NOT EXACTLY "0", then it was not a float. Hence the FIXME.
I know that @fgalan was struggling with this very same problem not so long ago ...
Help, please ... :-)